<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1068</width>
    <height>583</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Noto Sans Mono</family>
    <weight>50</weight>
    <bold>false</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout" stretch="5,2">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>744</width>
        <height>567</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout" columnminimumwidth="1,1">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item row="0" column="0">
        <widget class="CvWidget" name="cv_raw" native="true"/>
       </item>
       <item row="0" column="1">
        <widget class="CvWidget" name="cv_intermediate" native="true"/>
       </item>
       <item row="1" column="0">
        <widget class="CvWidget" name="cv_contours" native="true"/>
       </item>
       <item row="1" column="1">
        <widget class="CvWidget" name="cv_product" native="true"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QComboBox" name="mode">
       <item>
        <property name="text">
         <string>run</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>train</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>debug</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QStackedWidget" name="stacked">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="RUN">
        <property name="font">
         <font>
          <family>Noto Sans Mono</family>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="leftMargin">
          <number>8</number>
         </property>
         <property name="topMargin">
          <number>10</number>
         </property>
         <item>
          <widget class="QRadioButton" name="radioButton">
           <property name="text">
            <string>Remove the background</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="raw">
           <property name="text">
            <string>raw</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="intermediate">
           <property name="text">
            <string>intermediate</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="product_2">
           <property name="text">
            <string>contours</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="product">
           <property name="text">
            <string>product</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1">
           <item>
            <widget class="QLabel" name="label_result">
             <property name="font">
              <font>
               <family>Noto Sans Mono</family>
               <pointsize>15</pointsize>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Result:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="result_display">
             <property name="font">
              <font>
               <family>Noto Sans Mono</family>
               <pointsize>15</pointsize>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="TRAIN">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QLabel" name="data_label">
             <property name="text">
              <string>label</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="label_line">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>delay</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_freeze">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>Freeze</string>
             </property>
             <property name="autoDefault">
              <bool>false</bool>
             </property>
             <property name="default">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QPushButton" name="btn_adddata">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Add Data</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_savedata">
             <property name="text">
              <string>Save Data</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_save">
           <property name="text">
            <string>save path</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="0,0">
           <item>
            <widget class="QLineEdit" name="path_line">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_savesvm">
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btn_train">
           <property name="text">
            <string>Start Training</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="DEBUG">
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,0">
           <item>
            <widget class="QLabel" name="h_min">
             <property name="text">
              <string>hmin</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>0</number>
             </property>
             <property name="sliderPosition">
              <number>0</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="hmin">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1,0">
           <item>
            <widget class="QLabel" name="s_min">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>smin</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_2">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="smin">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="v_min">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>vmin</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_3">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="vmin">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="h_max">
             <property name="text">
              <string>hmax</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_6">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="sliderPosition">
              <number>255</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="hmax">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>255</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="s_max">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>smax</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_4">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="sliderPosition">
              <number>255</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="invertedAppearance">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="smax">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>255</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="v_max">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>vmax</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider_5">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="sliderPosition">
              <number>255</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="vmax">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>255</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_exit">
       <property name="text">
        <string>Exit</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CvWidget</class>
   <extends>QWidget</extends>
   <header>cvwidget.h</header>
   <container>1</container>
   <slots>
    <slot>toggleDisplay(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>btn_exit</sender>
   <signal>released()</signal>
   <receiver>Form</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1060</x>
     <y>575</y>
    </hint>
    <hint type="destinationlabel">
     <x>568</x>
     <y>494</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hmin</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>smin</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_3</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vmin</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_4</sender>
   <signal>valueChanged(int)</signal>
   <receiver>smax</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_5</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vmax</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_6</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hmax</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>858</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_3</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_6</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_4</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_5</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form</receiver>
   <slot>updateValue()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mode</sender>
   <signal>activated(int)</signal>
   <receiver>stacked</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1060</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>1060</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>raw</sender>
   <signal>toggled(bool)</signal>
   <receiver>cv_raw</receiver>
   <slot>toggleDisplay(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1054</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>171</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>intermediate</sender>
   <signal>toggled(bool)</signal>
   <receiver>cv_intermediate</receiver>
   <slot>toggleDisplay(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1054</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>678</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>product</sender>
   <signal>toggled(bool)</signal>
   <receiver>cv_product</receiver>
   <slot>toggleDisplay(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1054</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>631</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>Form</receiver>
   <slot>toggleBackground(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1054</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_freeze</sender>
   <signal>released()</signal>
   <receiver>Form</receiver>
   <slot>displayFreeze()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>991</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>746</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_adddata</sender>
   <signal>released()</signal>
   <receiver>Form</receiver>
   <slot>setDataAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>807</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>649</x>
     <y>-4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>label_line</sender>
   <signal>returnPressed()</signal>
   <receiver>Form</receiver>
   <slot>setLabelName()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>864</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>764</x>
     <y>-9</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>product_2</sender>
   <signal>toggled(bool)</signal>
   <receiver>cv_contours</receiver>
   <slot>toggleDisplay(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>847</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>418</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>changeIndex(int)</signal>
  <signal>changeOption(list)</signal>
  <slot>stop()</slot>
  <slot>updateValue()</slot>
  <slot>displayFreeze()</slot>
  <slot>toggleBackground(bool)</slot>
  <slot>setDataAction()</slot>
  <slot>setLabelName()</slot>
 </slots>
</ui>
